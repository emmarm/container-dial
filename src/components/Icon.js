import React from 'react';
import PropTypes from 'prop-types';
import styled from 'react-emotion';

const I = styled('i')({
  color: 'white',
  fontSize: 28
});

const Icon = ({ icon }) => <I className="material-icons">{icon}</I>;

Icon.propTypes = {
  icon: PropTypes.string
};

export default Icon;
