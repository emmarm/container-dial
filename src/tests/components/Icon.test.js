import React from 'react';
import { shallow } from 'enzyme';

import Icon from '../../components/Icon';

describe('Icon component', () => {
  const props = {
    icon: 'add'
  };
  const wrapper = shallow(<Icon {...props} />);

  it('matches snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('renders an icon', () => {
    const icon = wrapper.find('Styled(i)');
    expect(icon.length).toBe(1);
    const iconText = icon
      .dive()
      .find('i')
      .text();
    expect(iconText).toBe(props.icon);
  });

  it('applies padded prop if supplied', () => {
    const wrapperPadded = shallow(<Icon {...props} padded />);
    expect(wrapperPadded.prop('padded')).toBeTruthy();
  });
});
