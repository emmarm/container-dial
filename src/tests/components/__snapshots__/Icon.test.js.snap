// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon component matches snapshot 1`] = `
<Styled(i)
  className="material-icons"
>
  add
</Styled(i)>
`;
